apps/site/pages/docs/contextual-images.mdx

```diff
@@ -0,0 +1,123 @@
+---
+title: 'Using specific product images on different sections'
+sidebar_label: 'Contextual Images'
+---
+
+<header>
+
+# Contextual Images
+
+</header>
+
+Sometimes store owners want images to appear only in specific contexts, such as a product image that should appear only on the SKU Selector, not on the gallery.
+
+## How to use contextual images
+
+### 1. Use Labels on VTEX Catalog
+
+On VTEX's catalog, you should set the Label field of the product image to a context name. The only labels with special meaning are `generic` and `skuvariation`. If you label an image with `generic` you won't be able to filter for that individual label, you will always end up with the full set of images. 
+
+In this example we're going to [add the `gallery` label](https://help.vtex.com/en/tracks/catalog-101--5AF0XfnjfWeopIFBgs3LIQ/17PxekVPmVYI4c3OCQ0ddJ#adding-an-image-to-the-sku) to images that we want to display on the PDP's Image Gallery.
+
+### 2. Query for the correct images on their context
+
+Add fragments to both `ServerProductQuery` and `ClientProductQuery`:
+
+```js
+import { gql } from '@faststore/core/api'
+ 
+export const fragment = gql`
+  fragment ServerProduct on Query {
+    product(locator: $locator) {
+      galleryImages: image(context: "gallery", limit: 3) {
+        url
+        alternateName
+      }
+    }
+  }
+`
+```
+
+```js
+import { gql } from '@faststore/core/api'
+ 
+export const fragment = gql`
+  fragment ClientProduct on Query {
+    product(locator: $locator) {
+      galleryImages: image(context: "gallery", limit: 3) {
+        url
+        alternateName
+      }
+    }
+  }
+`
+```
+
+Note that we are only selecting three images by passing the `limit: 3` argument. If we omit this argument, all images labeled `gallery` will be returned. For backward compatibility, all the images are returned whenever a context is not found as a Label on a Product's images.
+
+### 3. Override the ImageGallery
+
+Create the `src/components/overrides/ProductDetails.tsx` file and add the following content to it:
+
+```tsx
+import { SectionOverride } from "@faststore/core";
+
+import { usePDP } from "@faststore/core";
+import { Image_unstable as Image } from "@faststore/core/experimental";
+
+import { ImageGallery, ImageGalleryViewer } from "@faststore/ui";
+import { useState } from "react";
+
+const SECTION = "ProductDetails" as const;
+
+const ImageComponent = ({ url, alternateName }: {url: string, alternateName?: string}) => {
+  return <Image src={url} alt={alternateName} width={68} height={68}/>
+}
+
+const override: SectionOverride = {
+  section: SECTION,
+  components: {
+    __experimentalImageGallery: {
+      Component: () => {
```

Comment:
Can't we extract this to `CusttomImageGallery` component? 🙈

Reply:
Are you afraid that the guide showing the component inline in this way will incentivize people to do the same in their stores? If that's the case I'll move it!

Reply:
yeah 🙈 Because in the others guides we're always recommending them to create a new component file...